prov/sockets/src/sock_ctx.c

```diff
@@ -71,6 +71,17 @@ struct sock_rx_ctx *sock_rx_ctx_alloc(const struct fi_rx_attr *attr,
 
 void sock_rx_ctx_free(struct sock_rx_ctx *rx_ctx)
 {
+	struct dlist_entry *entry;
+	struct sock_rx_entry *rx_buffered;
+
+	/* free any remaining buffered entries */
+	while (rx_ctx->rx_buffered_list.next != &rx_ctx->rx_buffered_list) {
+		entry = rx_ctx->rx_buffered_list.next;
+		rx_ctx->rx_buffered_list.next = entry->next;
+		rx_buffered = container_of(entry, struct sock_rx_entry, entry);
+		free(rx_buffered);
+	}
```

Comment:
Can you change this to use !dlist_empty() and dlist_pop_front()?

Reply:
Sure. Could you check whether I got it correctly?

Reply:
Looks good.  Can you split the line closer to 80 characters?  :)

Reply:
done.

Reply:
thanks!